### ISSUE TITLE
Allow API keys to be set by an environment variable

### ISSUE THREAD

**Issue by sloria** [asottile/all-repos#79]
Use case: I'd like to add all-repos.json to source control without committing my API_KEY.

```json
    "source_settings":  {
        "api_key": "$GITHUB_API_KEY",
        "org": "marshmallow-code"
    }
```

**Comment by asottile** [comment]
I don't really want to support arbitrary templating / complex configuration loading schemes (it's the reason json is chosen as the configuration format)

**Comment by sloria** [comment]
Fair enough. What about an additional option then, e.g. `"api_key_var": "GITHUB_API_KEY"`, that is mutually exclusive of `"api_key"`?

**Comment by asottile** [comment]
feels like that'll open a bunch of new edge cases and error cases -- I'd really rather keep this simple, I don't think that checking in the configuration is the right answer here.

If you want templating, you could always implement it with `sed` or such:

```bash
sed "s/API_KEY/${API_KEY}/g" all-repos.template.json > all-repos.json
```

check in the template and a little wrapper script / make target / whatever and you're good to go

**Comment by asottile** [comment]
plus you'll hit this wall with git-managed files: https://github.com/asottile/all-repos/blob/252fe6f5488df3482a68d6d41e842ea868eb4603/all_repos/config.py#L38
